apps/dotcom/sync-worker/src/TLDrawDurableObject.ts

```diff
@@ -342,6 +343,12 @@ export class TLDrawDurableObject extends DurableObject {
 				if (!auth && !file.shared) {
 					return closeSocket(TLSyncErrorCloseEventReason.NOT_AUTHENTICATED)
 				}
+				if (auth?.userId) {
+					const rateLimited = await isRateLimited(this.env, auth?.userId)
+					if (rateLimited) {
+						return closeSocket(TLSyncErrorCloseEventReason.FORBIDDEN)
```

Comment:
Not sure if we should introduce a new event reason here?

Reply:
yeah we should. `TLSyncErrorCloseEventReason.COOL_YOUR_JETS_HOMIE`

Reply:
(or just `TLSyncErrorCloseEventReason.RATE_LIMITED`)